include/stl2/view/take_exactly.hpp

```diff
@@ -88,10 +93,31 @@ STL2_OPEN_NAMESPACE {
 			constexpr iter_difference_t<iterator_t<Base>> size() const noexcept { return n_; }
 			constexpr bool empty() const noexcept { return n_ == 0; }
 		};
+
+		template <InputRange R>
+		take_exactly_view(R&& base, iter_difference_t<iterator_t<R>> n)
+			-> take_exactly_view<all_view<R>>;
 	} // namespace ext
 
 	template <class V>
 	struct enable_view<ext::take_exactly_view<V>> : std::true_type {};
+
+	namespace view::ext {
+		struct __take_exactly_fn : detail::__pipeable<__take_exactly_fn> {
+			template <class V>
+			auto operator()(V&& view, iter_difference_t<iterator_t<V>> const n) const
+			STL2_NOEXCEPT_REQUIRES_RETURN
+			(
+				__stl2::ext::take_exactly_view(std::forward<V>(view), n)
```

Comment:
s/b `__stl2::ext::take_exactly_view(all(std::forward<V>(view), n))` to convert `ViewableRange`s to `View`s.

Reply:
Ibid